pkg/identityserver/gateway_registry.go

```diff
@@ -77,6 +78,18 @@ func (is *IdentityServer) createGateway(ctx context.Context, req *ttnpb.CreateGa
 		req.FrequencyPlanIDs = []string{req.FrequencyPlanID}
 	}
 
+	if req.LBSLNSSecret != nil {
+		if is.config.Gateways.EncryptionKeyID == "" {
+			return nil, errGatewaySecretEncryptionKey
```

Comment:
This doesn't make sense. If it's not configured, we shouldn't return an error of type "not found". Perhaps make an `EncryptionKeyID` of `""` indicate "no encryption"? I think we also have that in the device registries.

Reply:
Should we allow unencrypted values to be stored in a field where we specifically say it's encrypted at rest?

Reply:
I don't think that "encrypted at rest" is relevant at the API level. In most deployments (including TTI Cloud) the entire database is encrypted at rest. This is just another layer of encryption that operators can choose to enable, just as with the root keys in the device registries.